regparser/web/jobs/views.py

```diff
@@ -148,3 +194,172 @@ def retrieve(self, request, *args, **kwargs):
+        Overrides the method from ``BaseViewList`` in order to pass the
+        arguments appropriate for the ``pipeline`` command.
+
+        Side Effects
+            Runs the ``pipeline`` command.
+
+        :arg dict validated_data: Incoming data from the POST that's already
+        been validated by the serializer.
+
+        :rtype: list[str]
+        :returns: The components of the argument string in list form.
+        """
+        return [
+            "pipeline",
+            str(validated_data["cfr_title"]),
+            str(validated_data["cfr_part"]),
+            eregs_site_api_url,
+        ]
+
+    def post(self, request, *args, **kwargs):
+        return self.create(request, *args, **kwargs)
+
+
+class PipelineJobViewInstance(JobViewInstance):
+    queryset = PipelineJob.objects.all()
+    serializer_class = PipelineJobSerializer
+    sub_path = "pipeline/"
+
+
+class ProposalPipelineJobViewList(BaseViewList, JobViewList):
+    queryset = ProposalPipelineJob.objects.all()
+    serializer_class = ProposalPipelineJobSerializer
+    sub_path = "proposal-pipeline/"
+
+    def build_eregs_args(self, validated_data):
+        """
+        Overrides the method from ``BaseViewList`` in order to pass the
+        arguments appropriate for the ``proposal_pipeline`` command.
+
+        Impure
+            Reads the contents of the proposal file from the filesystem (in
+            future, the DB, but impure either way).
+        Side Effects
+            Runs the ``proposal_pipeline`` command.
+
+        :arg dict validated_data: Incoming data from the POST that's already
+        been validated by the serializer.
+
+        :rtype: list[str]
+        :returns: The components of the argument string in list form.
+        """
+        reg_file = RegulationFile.objects.get(
+            hexhash=validated_data["file_hexhash"])
+        # TODO: This is a total hack; we should not be storing the contents in
+        # the DB but reading the file from the filesystem. Only doing this
+        # temporarily before changing the proposal_pipeline command to work
+        # differently.
+        path = reg_file.file.storage.path(reg_file.file.name)
+        return [
+            "proposal_pipeline",
+            path,
+            eregs_site_api_url
+        ]
+
+    def post(self, request, *args, **kwargs):
+        return self.create(request, *args, **kwargs)
+
+
+class ProposalPipelineJobViewInstance(JobViewInstance):
+    queryset = ProposalPipelineJob.objects.all()
+    serializer_class = ProposalPipelineJobSerializer
+    sub_path = "proposal-pipeline/"
+
+
+class FileUploadView(mixins.ListModelMixin, mixins.CreateModelMixin,
+                     generics.GenericAPIView):
+    parser_classes = (FileUploadParser, MultiPartParser)
+    parser_classes = (MultiPartParser,)
+    serializer_class = FileUploadSerializer
+    queryset = RegulationFile.objects.all()
+    lookup_field = "hexhash"
+    size_limit = 100000000  # Arbitrary 100MB limit.
```

Comment:
Actually, reading below, we might not be able to load 100M in memory (our instances tend to be in the 512M range); would 10M make more sense?